insights/parsers/tests/test_tomcat_virtual_dir_context.py

```diff
@@ -0,0 +1,57 @@
+import pytest
+
+from insights.parsers.tomcat_virtual_dir_context import TomcatVirtualDirContext
+from insights.tests import context_wrap
+from insights.parsers import ParseException
+
+FOUND_1 = """
+/usr/share/tomcat/conf/server.xml:    <Resources className="org.apache.naming.resources.VirtualDirContext"
+""".strip()
+
+FOUND_2 = """
+/usr/share/tomcat/conf/server.xml:    <Resources className="org.apache.naming.resources.VirtualDirContext"
+/usr/share/tomcat6/webapps/whatever/META-INF/context.xml:className="org.apache.naming.resources.VirtualDirContext"
+""".strip()
+
+NOT_FOUND = """
+""".strip()
+
+ERRORS_1 = """
+/bin/grep: No such file or directory
+"""
+
+ERRORS_2 = """
+garbage garbage
+"""
+
+
+def test_tomcat_virtual_dir_context_found():
+    tomcat_virtual_dir_context = TomcatVirtualDirContext(context_wrap(FOUND_1))
+    assert tomcat_virtual_dir_context.has_virtual_dir_context is True
+    assert len(tomcat_virtual_dir_context.lines) == 1
+    assert tomcat_virtual_dir_context.lines[0] == ('/usr/share/tomcat/conf/server.xml',
+                                                   '    <Resources className="org.apache.naming.resources.VirtualDirContext"')
+
+    tomcat_virtual_dir_context = TomcatVirtualDirContext(context_wrap(FOUND_2))
+    assert tomcat_virtual_dir_context.has_virtual_dir_context is True
+    assert len(tomcat_virtual_dir_context.lines) == 2
+    assert tomcat_virtual_dir_context.lines[0] == ('/usr/share/tomcat/conf/server.xml',
+                                                   '    <Resources className="org.apache.naming.resources.VirtualDirContext"')
+    assert tomcat_virtual_dir_context.lines[1] == ('/usr/share/tomcat6/webapps/whatever/META-INF/context.xml',
+                                                   'className="org.apache.naming.resources.VirtualDirContext"')
+
+
+def test_tomcat_virtual_dir_context_not_found():
```

Comment:
How about raising a "ParseException" or "SkipExecption" for empty input content (NOT_FOUND)? In the case of raising exception, the `.has_virtual_dir_context` attribute is not needed since this parser will not be triggered. 

That is to say, for the cases, this parser is triggered the `.has_virtual_dir_context` must be `True`.

And by doing that, the plugin's code would be simpler.

Reply:
`SkipExecption` seems like a good idea. It is a bit outside my usual thinking about parsers, but makes sense. Will be updated.